starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,501 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(a, b) {
+  if (a > b) {
+    return a;
+  }
+  return b;
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(words) {
+  if (words.length == 0) {
+    return null;
+  }
+
+  let longestWord = "";
+  words.forEach(function(word) {
+    if (word.length > longestWord.length) {
+      longestWord = word;
+    }
+  });
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(nums) {
+  let sum = 0;
+  for (const num of nums) {
+    sum += num;
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(nums) {
+  if (nums.length == 0) {
+    return null;
+  }
+  return sumArray(nums) / nums.length;
+}
+
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(words) {
+  let wordLengths = [];
+  for (const word of words) {
+    wordLengths.push(word.length);
+  }
+  return averageNumbers(wordLengths);
+}
+
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(words) {
+  let newArray = [];
+  words.forEach(function(word) {
+    // add to new array if it does not already exist
+    if (newArray.indexOf(word) == -1) {
+      newArray.push(word);
+    }
+  });
+  return newArray;
+}
+
 // Iteration #6: Find elements
 const wordsFind = [
-  'machine',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'truth',
-  'disobedience'
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
 ];
 
+function doesWordExist(words, searchWord) {
+  if (words.indexOf(searchWord) == -1) {
+    return false;
```

Comment:
Nice! 💯 have a look at the `includes` method which will return a boolean - again also could just make your code shorter. :)